### PR TITLE
feat: add API descriptor with API_SPECTRAL with AsyncAPI/OpenAPI support

### LINKED ISSUE #3387: Support AsyncAPI linting using Spectral
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. I'm always frustrated when […]

Currently, AsyncAPI definitions are not ignored/not linted by Spectral.

**Describe the solution you'd like**

There is a linter OPENAPI_SPECTRAL with the config file `.openapirc.yml`.
The default file shipped with MegaLinter contains
```yaml
extends: spectral:oas
```

However, Spectral supports OpenAPI **and** AsyncAPI.

```yaml
extends:
  - spectral:asyncapi
  - spectral:oas
```
could be used to validate both.

The recommended naming for the config file is `.spectral.yaml` according to their documentation.

However, the linter is called `OPENAPI_SPECTRAL` which sets the scope to OpenAPI only.

Similar to others like prettier, we could add an additional linter `ASYNCAPI_SPECTRAL`.
Either with a config file `.asyncapirc.yml` or using a shared config file `.spectral.yaml` as default.

Changing the default config filename would be fine as long as there is some backward compatibility
to custom `.openapirc.yml` files with repos.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by pjungermann**
Add `API` descriptor as modified copy of the `OPENAPI` descriptor. Deprecate the contained linter `OPENAPI_SPECTRAL`.

Add `.spectral.yaml` as default config file to `TEMPLATE/` as modified copy of `.openapirc.yml` with rulesets for AsyncAPI and OpenAPI (previously, only OpenAPI).

Fixes #3387

## Proposed Changes

1. Adds `API_SPECTRAL` as modified copy of `OPENAPI_SPECTRAL`.
2. Supports AsyncAPI and OpenAPI (filenames, file contents, etc.).
3. Uses `.spectral.yaml` as config filename and provides a default file at `TEMPLATE/` with rulesets for AsyncAPI and OpenAPI.
4. Deprecates `OPENAPI_SPECTRAL` (and therefore, `OPENAPI`)

### Author/Contributor
- [x] Add entry to the [CHANGELOG](https://github.com/oxsecurity/megalinter/blob/main/CHANGELOG.md) listing the change and linking to the corresponding issue (if appropriate)
- [x] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [ ] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -206,6 +206,7 @@ WORKDIR /node-deps
 RUN npm --no-cache install --ignore-scripts --omit=dev \
                 @salesforce/cli \
                 typescript \
+                @stoplight/spectral-cli \
```
Comment (echoix) [inline Dockerfile:209]: @nvuillam idea for improvement here: sort the required packages, to not have the order changed uselessly like here.

**Comment by echoix** [comment]
I think there's a bit of CI issues, in part with the new eslint release, so it might not get merged today, but I think it's good overall here ;)

File: Dockerfile (hunk 1 above)
Comment (nvuillam) [reply Dockerfile:209]: 100% agreed, can be done in another PR :)

**Comment by pjungermann** [comment]
I guess I cannot make lychee happy

```
  Errors in README.md
  ✗ [404] https://github.com/oxsecurity/megalinter/tree/main/docs/descriptors/api_spectral.md | Failed: Network error: Not Found
  ✗ [404] https://github.com/oxsecurity/megalinter/tree/main/docs/descriptors/api.md | Failed: Network error: Not Found
```

as these files don't exist at the main branch, yet.

**Comment by echoix** [comment]
However three tests for api spectral report failure : https://github.com/oxsecurity/megalinter/actions/runs/8694444485/job/23843361154#step:11:1

**Comment by nvuillam** [comment]
@pjungermann test classes are failing ^^

**Review by echoix — APPROVED** [review]
I won't merge now, since two issues on GitHub status makes the workflows on my other repo all fail since a couple hours. Now tests don't fail. The only error is the docs not yet published for the contents of this PR

**Comment by echoix** [comment]
In fact it will take @nvuillam to force the merge later on since it is required but fails.

**Comment by pjungermann** [comment]
Thank you!

**Review by nvuillam — APPROVED** [review]

**Pull request merged**